MapboxCoreNavigation/RouteController.swift

```diff
@@ -316,7 +316,7 @@ open class RouteController: NSObject {
         sendOutstandingFeedbackEvents(forceAll: true)
         suspendNotifications()
         
-        let monitorBatteryValue = delegate?.routeControllerWillDisableBatteryMonitoring?(self) ?? false
+        let monitorBatteryValue = delegate?.routeControllerShouldDisableBatteryMonitoring?(self) ?? false
```

Comment:
Now the semantics are still wrong: If I return `false` because it shouldn't be disabled, it will actually be disabled and the other way around.

Reply:
Yes; it's not correct to pass this value into the next line. The heuristic should look like:

 - If the delegate doesn't implement `routeControllerShouldDisableBatteryMonitoring()`, go ahead and disable it.
 - If the delegate implements `routeControllerShouldDisableBatteryMonitoring()` and returns true, disable it.
 - If the delegate implements `routeControllerShouldDisableBatteryMonitoring()` and returns false, don't disable it.